package.json

```
@@ -30,7 +30,7 @@
     "docs": "jsdoc -c .jsdoc.js",
     "predocs": "npm run compile",
     "lint": "gts fix",
-    "samples-test": "cd samples/ && npm link ../ && npm test && cd ../",
+    "samples-test": "cd samples/ && rm -rf node_modules package-lock.json && npm link ../ && npm install && npm test && cd ../",
```

removing `node_modules` and `package-lock.json` as part of `sample-tests` seems weird to me. In CI/CD these files don't exist, in a local environment, this may result in you needing to reinstall dependencies repeatedly as you test.

---

@AVaksman is the idea to force the latest code in src/ to compile and be relinked? Does just this work?:

```suggestion
    "samples-test": "cd samples/ && npm link ../ && npm install && npm test",
```

npm install should noop relatively quickly if nothing needs to be done. Removing package-lock.json for the user makes sense, although is there a way to disable package-lock.json on a repo level?

---

I am strongly opposed to hiding `rm -rf` in scripts.

---

adding `npm install` to the script should be enough. `rm -rf` is way to harsh.
to disable package-lock.json on a repo level an `.npmrc` with `package-lock=false` could be added.
WDYT?

---

I'd be okay with disabling package-lock.json. There's not a good time I can think of that a maintainer should be okay using stale dependencies to run the tests against.